### ISSUE TITLE
Bubled PGP keys: How to prevent forgery? How do you verify PRs?

### ISSUE THREAD

**Issue by rugk** [tasn/webext-signed-pages#9]
I've made another project aware [of this extension](https://github.com/threema-ch/threema-web/issues/427) and one question about the bundled PGP keys came up:

> Furthermore, how would they prevent an attacker from simply sending a pull request that replaces the public key, claiming to work at that company?

So how would you do?

This is more about the process when/how you accept a PR? IMHO, you need to verify the ownership of the domain (e.g. just let them sign some random stuff you give them and say they should put that on the website), don't you? At least that would be the minimum…
Of course, it defeats the whole purpose as an attacker, who overtook a server, can also verify ownership of the domain, so you have to take this into account and deliberately delay the PR merging or so, and probably need to let them verify it again, so that it was not a "short time the server was overtaken".

Or you require PRs to be official GitHub  accounts of that company?
Or some stuff involving Keybase?

**Comment by tasn** [comment]
Thanks for opening this. It's not something I've given any thought to, but it's a very important matter and needs documentation.
I like using Keybase because it brings in some sort of assurances. Same goes with PGP keyservers. Less so with Github. I think I'd require either that, or maybe some sort of a triple confirmation mechanism, e.g, an email from the domain (and back to them), a tweet from their account, and maybe a file on the server or a mention on their website (which is a good so their users know).

**Comment by rugk** [comment]
Note that PGP keyserver do **not* bring any assurance. Anyone can upload a PGP key there, so nothing is verified and that does not work here.

**Comment by rugk** [comment]
BTW, you must not only cover the "initial addition", but also the case when the company may want to update the key as it is lost.

**Comment by tasn** [comment]
Yeah, I know, I had some sort of checking you trust a key based on the web of trust idea in mind, but in second thought, I'm not sure how that would work exactly.

Initial addition and change would probably be the same (or a very similar process). With that being said, people should keep their PGP keys safe. Changing keys is quite a drastic event that should be avoided.
